synapse/crypto/context_factory.py

```diff
@@ -32,8 +32,11 @@ def __init__(self, config):
     @staticmethod
     def configure_context(context, config):
         try:
-            _ecCurve = _OpenSSLECCurve(_defaultCurveName)
-            _ecCurve.addECKeyToContext(context)
+            # This was removed in https://github.com/twisted/twisted/pull/928
+            # _ecCurve = _OpenSSLECCurve()
+            _evCurve = crypto.get_elliptic_curve(_defaultCurveName)
```

Comment:
this appears to have been introduced in pyopenssl 0.15 - so at the very least we need to bump the requirement in python_dependencies.py.

However, there is a problem in that debian jessie has 0.14. There is a newer version in backports though - @erikjohnston is it ok to rely on things in backports? 

Otherwise we might have to do some hackery dependent on the Twisted version.

Reply:
> However, there is a problem in that debian jessie has 0.14. There is a newer version in backports though - @erikjohnston is it ok to rely on things in backports?

It is, though I think we do some foo to host the backported packages we need on our repo. I've just had a look at it seems that we're already hosting v0.16, so this is fine.